src/index.js

```diff
@@ -29,6 +29,16 @@ module.exports = class BroccoliDebug extends Plugin {
   constructor(node, labelOrOptions) {
     let options = processOptions(labelOrOptions);
 
+    // this is used to avoid creating extraneous broccoli nodes when they are
+    // not going to be used for debugging purposes (e.g. `BROCCOLI_DEBUG=*`
+    // isn't present or doesn't apply to the current label)
+    //
+    // note: classes can only return an object or undefined from their
+    // constructor, hence the guard for typeof object
+    if (typeof node === 'object' && !options.force && !match(options.label)) {
+      return node;
+    }
```

Comment:
@rwjblue should `node` being a string also be covered here? See https://github.com/broccolijs/broccoli-concat/pull/138#discussion_r222981700

Reply:
Yes, seems good! Lets change this:

1. Move broccoli-source to a dep (from devDeps)
2. If node is a string return a WatchedDir